src/magic_folder/snapshot.py

```diff
@@ -0,0 +1,247 @@
+    """
+    key_fname = "magicfolder_{}.privkey".format(magic_folder_name)
+    path = config.get_config_path("private", key_fname)
+    keydata_base64 = local_author.signing_key.encode(encoder=Base64Encoder)
+    author_data = {
+        "author_name": local_author.name,
+        "author_private_key": keydata_base64,
+    }
+    with open(path, "w") as f:
+        json.dump(author_data, f)
+
+
+def create_local_author_from_config(config, name=None):
+    """
+    :param config: a Tahoe config instance (created via `allmydata.client.read_config`)
+
+    :param name: which Magic Folder to use (or 'default')
+
+    :returns: a LocalAuthor instance from our configuration
+    """
+    # private-keys go in "<node_dir>/private/magicfolder_<name>.privkey"
+    # to mirror where the sqlite database goes
+    if name is None:
+        name = "default"
+    nodedir = config.get_config_path()
+    magic_folders = load_magic_folders(nodedir)
+    if name not in magic_folders:
+        raise RuntimeError(
+            "No magic-folder named '{}'".format(name)
+        )
+
+    # we will always have authorship information for this
+    # magic-folder; "legacy" magic-folders will go through "tahoe
+    # migrate" first and have an author created.
+
+    author_raw = config.get_private_config("magicfolder_{}.privkey".format(name))
+    author_data = json.loads(author_raw)
+
+    return LocalAuthor(
+        name=author_data[u"author_name"],
+        signing_key=SigningKey(
+            author_data[u"author_private_key"],
+            encoder=Base64Encoder,
+        ),
+    )
+
+
+@attr.s
+class LocalSnapshot(object):
+    name = attr.ib()
+    author = attr.ib()
+    metadata = attr.ib()
+    content_path = attr.ib()  # full filesystem path to our stashed contents
+    parents_local = attr.ib()  # LocalSnapshot instances
+
+    # once we do uploads / downloads and have RemoteSnapshots, we will
+    # also have those kind of parents too:
+    # parents_remote = attr.ib()
+    # ..and need to add methods here to count and async get parents
+
+    def get_content_producer(self):
+        """
+        :returns: an IBodyProducer that gives you all the bytes of the
+            on-disc content. Raises an error if we already have a
+            capability. Note that this data will have been stashed previously.
+        """
+        return FileBodyProducer(
+            open(self.content_path, "rb")
+        )
+
+    def _get_synchronous_content(self):
+        """
+        For testing only.
+        :returns: the content immediately
+        """
+        with open(self.content_path, "rb") as f:
+            return f.read()
+
+
+@inlineCallbacks
+def create_snapshot(name, author, data_producer, snapshot_stash_dir, parents=None):
+    """
+    Creates a new LocalSnapshot instance that is in-memory only. All
+    data is stashed in `snapshot_stash_dir` before this function
+    returns.
+
```

Comment:
Missing docstring for the `name` parameter.

Reply:
fixed, thanks